processing/src/test/java/io/druid/query/topn/TopNQueryTest.java

```diff
@@ -74,4 +74,22 @@ public void testQuerySerialization() throws IOException
     Assert.assertEquals(query, serdeQuery);
   }
 
+  @Test
+  public void testQuerySerdeWithAlphaNumericTopNMetricSpec() throws IOException{
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(dataSource)
+        .granularity(allGran)
+        .dimension(marketDimension)
+        .metric(new AlphaNumericTopNMetricSpec(null))
+        .threshold(2)
+        .intervals(fullOnInterval)
+        .aggregators(Lists.<AggregatorFactory>newArrayList(commonAggregators))
+        .build();
+
+    String json = jsonMapper.writeValueAsString(query);
```

Comment:
for serde tests we should parse an actual json. Having a round-trip test is not enough, it only tells us that the fields we write parse correctly, but it doesn't tell us that we actually write or read all the fields.